java/core/src/main/java/com/microsoft/bond/NullableBondType.java

```diff
@@ -120,6 +120,20 @@ protected final TValue deserializeValue(TaggedDeserializationContext context) th
     }
 
     @Override
+    protected final TValue deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        if (count == 0) {
+            return null;
+        } else if (count == 1) {
```

Comment:
This makes me wonder if the code above for tagged protocols should also check for empty lists in payload. Such list would not be produced by serialization (since empty lists are omitted) but if we do get this in payload, throwing an exception complaining about "multiple elements" for an empty list sounds wrong.

What are your thoughts on this?